Mathlib/Topology/SubsetProperties.lean

```diff
@@ -1307,6 +1309,120 @@ theorem IsClosed.exists_minimal_nonempty_closed_subset [CompactSpace α] {S : Se
   rw [← this, compl_compl]
 #align is_closed.exists_minimal_nonempty_closed_subset IsClosed.exists_minimal_nonempty_closed_subset
 
+/-- A subset `s ⊆ α` is called **σ-compact** if it is the union of countably many compact sets. -/
+def IsSigmaCompact (s : Set α) : Prop :=
+  ∃ K : ℕ → Set α, (∀ n, IsCompact (K n)) ∧ ⋃ n, K n = s
+
+/-- The empty set is σ-compact. -/
+@[simp]
+lemma isSigmaCompact_empty : IsSigmaCompact (∅ : Set α) := by
+  use fun _ ↦ ∅
+  simp only [isCompact_empty, forall_const, iUnion_empty, and_self]
+
+/-- Compact sets are σ-compact. -/
+lemma IsCompact.isSigmaCompact {s : Set α} (hs : IsCompact s) : IsSigmaCompact s :=
+  ⟨fun _ => s, fun _ => hs, iUnion_const _⟩
+
+/-- Countable unions of compact sets are σ-compact. -/
+lemma isSigmaCompact_of_countable_compact {S : Set (Set α)} (hc : Set.Countable S)
```

Comment:
I think this should be named either `isSigmaCompact_sUnion_of_isCompact` (I don't think it's important to emphasize on the countability in the same), or `Set.Countable.isSigmaCompact_sUnion_of_isCompact` if you want to use dot notation.